staging/src/k8s.io/kubectl/pkg/cmd/cmd.go

```diff
@@ -173,21 +174,41 @@ func NewDefaultPluginHandler(validPrefixes []string) *DefaultPluginHandler {
 func (h *DefaultPluginHandler) Lookup(filename string) (string, bool) {
 	for _, prefix := range h.ValidPrefixes {
 		path, err := exec.LookPath(fmt.Sprintf("%s-%s", prefix, filename))
-		if err != nil || len(path) == 0 {
-			continue
+		if err != nil {
```

Comment:
Since 1.19 gives us a specific error indicator for the dot-path lookup, can we use that rather than string checks?

I think that would limit the changes in this file to:
```diff
- if err != nil || len(path) == 0 {
+ if shouldSkipOnLookPathErr(err) || len(path) == 0 {
```

then the <=go1.18 impl is:
```go
// +build !go1.19
...
func shouldSkipOnLookPathErr(err error) bool {
  return err != nil
}
```

and the 1.19+ impl is:
```go
// +build go1.19
...
func shouldSkipOnLookPathErr(err error) bool {
  return err != nil && !errors.Is(err, exec.ErrDot)
}
```